DREAM3DReviewFilters/FindMaskNeighbors.cpp

```diff
@@ -0,0 +1,392 @@
+{
+  int64_t modX = x + neighborhood[3 * index + 0];
+  int64_t modY = y + neighborhood[3 * index + 1];
+  int64_t modZ = z + neighborhood[3 * index + 2];
+
+  return (modX >= 0 && modX < dims[0]) && (modY >= 0 && modY < dims[1]) && (modZ >= 0 && modZ < dims[2]);
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+bool FindMaskNeighbors::validNeighbor2D(int64_t dims[2], int64_t neighborhood[16], size_t index, int64_t x, int64_t y)
+{
+  int64_t modX = x + neighborhood[2 * index + 0];
+  int64_t modY = y + neighborhood[2 * index + 1];
+
+  return (modX >= 0 && modX < dims[0]) && (modY >= 0 && modY < dims[1]);
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void FindMaskNeighbors::find_surfacefeatures()
+{
+  int64_t neighborhood[78] = { 1,  0, 0,  -1, 0, 0, 0, 1, 0,  0, -1, 0, 0, 0,  1,  0, 0, -1, 1, 1, 0,  -1, 1,  0, 1, -1, 0,  -1, -1, 0, 1,  0, 1,  1,  0,  -1, -1, 0,  1,
+                            -1, 0, -1, 0,  1, 1, 0, 1, -1, 0, -1, 1, 0, -1, -1, 1, 1, 1,  1, 1, -1, 1,  -1, 1, 1, -1, -1, -1, 1,  1, -1, 1, -1, -1, -1, 1,  -1, -1, -1 };
+  DataContainer::Pointer m = getDataContainerArray()->getDataContainer(getFeatureIdsArrayPath().getDataContainerName());
+
+  int64_t xPoints = static_cast<int64_t>(m->getGeometryAs<ImageGeom>()->getXPoints());
+  int64_t yPoints = static_cast<int64_t>(m->getGeometryAs<ImageGeom>()->getYPoints());
+  int64_t zPoints = static_cast<int64_t>(m->getGeometryAs<ImageGeom>()->getZPoints());
+
+  int64_t dims[3] = {xPoints, yPoints, zPoints};
+
+
+  int64_t zStride = 0, yStride = 0;
+  for(int64_t i = 0; i < zPoints; i++)
+  {
+    zStride = i * xPoints * yPoints;
+    for(int64_t j = 0; j < yPoints; j++)
+    {
+      yStride = j * xPoints;
+      for(int64_t k = 0; k < xPoints; k++)
+      {
+        int32_t gnum = m_FeatureIds[zStride + yStride + k];
+        if(!m_MaskNeighbors[gnum])
+        {
+          if (!m_Mask[yStride + k])
+          {
+
+            std::vector<int> validNeighborIndicies;
+
+            for (size_t n = 0; n < 8; n++)
+            {
+              if (validNeighbor(dims, neighborhood, n, k, j, i))
+              {
+                int64_t neighbor = ((j + neighborhood[2 * n + 1]) * dims[0]) + (k + neighborhood[2 * n + 0]);
+                if (m_Mask[neighbor])
+                {
+                  m_MaskNeighbors[gnum] = true;
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void FindMaskNeighbors::find_surfacefeatures2D()
+{
```

Comment:
Are we really finding sureface features or was this a repurposed method but just not ever renamed? Same for the 3D version of the method